crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -144,6 +161,111 @@ impl AuthorityStorePruner {
         Ok(())
     }
 
+    fn prune_checkpoints(
+        perpetual_db: &Arc<AuthorityPerpetualTables>,
+        checkpoint_db: &Arc<CheckpointStore>,
+        checkpoint_number: CheckpointSequenceNumber,
+        checkpoints_to_prune: Vec<(CheckpointSequenceNumber, CheckpointDigest)>,
+        checkpoint_content_to_prune: Vec<CheckpointContents>,
+        effects_to_prune: &Vec<TransactionEffects>,
+        metrics: Arc<AuthorityStorePruningMetrics>,
+    ) -> anyhow::Result<()> {
+        let _scope = monitored_scope("EffectsLivePruner");
+
+        let mut perpetual_batch = perpetual_db.objects.batch();
+        let transactions = checkpoint_content_to_prune
+            .iter()
+            .flat_map(|content| content.iter().map(|tx| tx.transaction));
+        for transaction_digest in transactions {
+            if let Some(next_digest) = transaction_digest.next_lexicographical() {
+                perpetual_batch.delete_range(
+                    &perpetual_db.transactions,
+                    &transaction_digest,
+                    &next_digest,
```

Comment:
wow, so we have to use range deletes instead of point deletes even what we are doing is actually a point delete? i am so confused as to how this is better for rocksdb!

Reply:
yeah, that's kind of confusing:) We do the same for the `objects` cf btw.
Point-delete is the same as another insertion with a tombstone for rocksdb and has dramatic perf implications.
Using range deletion as a point deletion may seem a bit hacky, but the main difference is that range deletions are stored in separate data structures(in both SST files and memtable) and can be disabled from the read path, so they won't impact reads. And eventually, the compaction job takes care of reclaiming the space

Reply:
ah, so this has the very interesting implication that this may not help us assess correctness very well, if the deleted elements are still visible.

Reply:
eventually, the data is going to be removed, and we can enforce immediate deletion in [simtest](https://github.com/MystenLabs/sui/pull/12134). Maybe we can do something similar for FN testing too, but only for short period of time cause running it so is very expensive